src/main/java/frc/robot/commands/IntakeSequence.java

```diff
@@ -1,54 +1,96 @@
 
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.geometry.Rotation2d;
 import edu.wpi.first.math.geometry.Transform2d;
 import edu.wpi.first.networktables.NetworkTableInstance;
 import edu.wpi.first.wpilibj2.command.ConditionalCommand;
 import edu.wpi.first.wpilibj2.command.InstantCommand;
 import edu.wpi.first.wpilibj2.command.SequentialCommandGroup;
 import edu.wpi.first.wpilibj2.command.WaitUntilCommand;
+import frc.robot.RobotContainer;
 import frc.robot.subsystems.IntakeLimelight.IntakeLimelight;
+import frc.robot.subsystems.ScoringLimelight.ScoringLimelight;
 import frc.robot.subsystems.arm.Arm;
 import frc.robot.subsystems.arm.Arm.ArmPosition;
 import frc.robot.subsystems.claw.Claw;
 import frc.robot.subsystems.climb.Climb;
 import frc.robot.subsystems.climb.Climb.ClimbPosition;
+import frc.robot.subsystems.drive.DriveSubsystem;
 import frc.robot.subsystems.elevator.Elevator;
 import frc.robot.subsystems.elevator.Elevator.ElevatorHeight;
+import frc.robot.utils.HPUtil;
 
 public class IntakeSequence extends SequentialCommandGroup {
-    private Optional<Transform2d> isTag = Optional.empty();
-    private double tagid = NetworkTableInstance.getDefault().getTable("limelight").getEntry("tid").getDouble(0);
-    public IntakeSequence(Claw claw, IntakeLimelight intakeLime, Arm arm, Elevator elevator, Climb climb) {
-        isTag = intakeLime.checkForTag();
-        SequentialCommandGroup moveArmElevatorClaw = new SequentialCommandGroup(
-            /*TODO do drive logic here*/
-            // the next two need to be done at the same time as the driving
+    public static Transform2d robotToTag;
+    public static Pose2d targetPose;
+  public IntakeSequence(
+      Claw claw, IntakeLimelight intakeLime, Arm arm, Elevator elevator, Climb climb, ScoringLimelight scoringLimelight, RobotContainer.Location location, DriveSubsystem drive) {
+    /* mechanical intake sequence - slightly modified version of what ruben had previously */
+    SequentialCommandGroup moveArmElevatorClaw =
+        new SequentialCommandGroup(
             new ConditionalCommand(
-                    new SequentialCommandGroup(
-                        new InstantCommand(() -> climb.setDesiredClimbPosition(ClimbPosition.CLEAR_OF_ARM)),
-                        new InstantCommand(() -> arm.setDesiredPosition(ArmPosition.INTAKE))
-                    ),
-                    new InstantCommand(() -> arm.setDesiredPosition(ArmPosition.INTAKE)), 
-                    () -> {return arm.isArmMoveable();}),
+                new SequentialCommandGroup(
+                    new InstantCommand(
+                        () -> climb.setDesiredClimbPosition(ClimbPosition.CLEAR_OF_ARM)),
+                    new WaitUntilCommand(()->climb.atDesiredPosition()),
+                    new InstantCommand(() -> arm.setDesiredPosition(ArmPosition.INTAKE))),
+                new InstantCommand(() -> arm.setDesiredPosition(ArmPosition.INTAKE)),
```

Comment:
Maybe I'm a stickler for this, but I always avoid duplicate typing of commands by just making one of them an empty InstantCommand and moving it outside the ConditionalCommand